### PR TITLE
parser: rework error handling

### PR DESCRIPTION
**Pull request by belochub**
Avoid using TryCatch introduced in
29b3e197ae4c33d1b9057b57731b4c422de764f3 by using MaybeLocal instead.

Refs: https://github.com/metarhia/jstp/pull/143

### CODE REVIEW

File: src/jsrs_parser.cc [hunk 1]
```diff
@@ -273,7 +269,7 @@ Local<Value> ParseBool(Isolate*    isolate,
     *size = 5;
   } else {
     THROW_EXCEPTION(TypeError, "Invalid format: expected boolean");
-    result = Undefined(isolate);
+    result = MaybeLocal<Value>();
```
Comment (aqrln) [inline src/jsrs_parser.cc]: Isn't this the very same value that `result` was initialized with at line 263?

Comment (belochub) [inline src/jsrs_parser.cc]: Yeah, right, I will remove redundant assignments, thanks for catching that 😃

File: src/jsrs_parser.cc [hunk 2]
```diff
@@ -90,16 +90,11 @@ Local<Value> Parse(Isolate* isolate, const String::Utf8Value& in) {
...
-    TryCatch trycatch(isolate);
-    result =
-        (kParseFunctions[type])(isolate, str + start_pos, end, &parsed_size);
-
-    if (trycatch.HasCaught()) {
-      trycatch.ReThrow();
-      return Undefined(isolate);
-    }
+  MaybeLocal<Value> result;
+  result = (kParseFunctions[type])(isolate, str + start_pos, end, &parsed_size);
```
Comment (lundibundi) [inline src/jsrs_parser.cc]: Just a nit but maybe it's better to make it into one statement? =)

File: src/jsrs_parser.cc [hunk 3]
```diff
@@ -698,20 +684,17 @@ Local<Value> ParseArray(Isolate*    isolate,
...
-        t = kParseFunctions[current_type](isolate,
-                                          begin + i,
-                                          end,
-                                          &current_length);
-        if (trycatch.HasCaught()) {
-          trycatch.ReThrow();
-          return Undefined(isolate);
-        }
+      MaybeLocal<Value> t;
+      t = kParseFunctions[current_type](isolate,
```
Comment (lundibundi) [inline src/jsrs_parser.cc]: Same as before, this one will event fit actually.

**Comment by aqrln** [comment]
Landed in a67612528d832ea9349e36093a6d3a941caaae22. Thanks!

**Pull request closed without merging**